server/service/orbit_client.go

```diff
@@ -146,6 +146,20 @@ func (oc *OrbitClient) SetOrUpdateDeviceToken(deviceAuthToken string) error {
 	return nil
 }
 
+// SetOrUpdateDeviceMappingEmail sends a request to the server to set or update the
+// device mapping email with the given value.
+func (oc *OrbitClient) SetOrUpdateDeviceMappingEmail(email string) error {
+	verb, path := "POST", "/api/fleet/orbit/device_mapping"
```

Comment:
Just a heads-up that following discussion with Noah, this is now a `PUT` (and the Request/Response structs have been renamed accordingly). I updated it in [my CLI PR](https://github.com/fleetdm/fleet/pull/15763/files#diff-6697275ba79dc39a42a714475e32a58103bbedf3f3de67b7183f979276049606R151) but I'm not sure which of our PRs will be merged last, you may want to rebase off of https://github.com/fleetdm/fleet/pull/15748 and update this.